packages/web3-eth/src/utils/transaction_builder.ts

```diff
@@ -60,25 +62,30 @@ export async function defaultTransactionBuilder<ReturnType = Record<string, unkn
 	web3Context: Web3Context<EthExecutionAPI & Web3NetAPI>;
 	privateKey?: HexString | Buffer;
 }): Promise<ReturnType> {
-	let populatedTransaction = { ...options.transaction } as unknown as Transaction;
+	let populatedTransaction = { ...options.transaction } as unknown as InternalTransaction;
```

Comment:
I don't think it's safe to assume that the transaction is `InternalTransaction` (a transaction with out hex string number and byte properties) if the parameter `transaction` has type `Record<string, unknown>`

Unless TypeScript has a fit about it, I think we should have `transaction` be of type [Transaction](https://github.com/ChainSafe/web3.js/blob/32b898477c455c8773e094e5b32a3d328a328ea4/packages/web3-eth/src/types.ts#L71) and we should format `populatedTransaction` to hex string

Reply:
Created an issue to address this https://github.com/ChainSafe/web3.js/issues/4892